File: src/Tui/AgentDisplayState.php
<?php

declare(strict_types=1);

namespace SugarCraft\Crush\Tui;

/**
 * Represents the live display state of a single agent in the status bar.
 *
 * Collected from the agent pool at render time so the status bar can
 * display name, operational status, current task, elapsed wall-clock
 * time, and accumulated token usage without the renderer needing to
 * know about internal agent state.
 */
final readonly class AgentDisplayState
{
    public function __construct(
        /** Agent name or role label, e.g. "coder-1", "reviewer-2". */
        public string $name,
        /** Human-readable operational status string shown in the bar. */
        public string $status,
        /** Short description of what the agent is currently doing. */
        public string $operation,
        /** Wall-clock seconds since the agent was started. */
        public int $elapsedSeconds,
        /** Total input + output tokens consumed so far. */
        public int $tokensUsed,
        /** Total cost in USD so far. */
        public float $costUsd,
    ) {}

    /**
     * Formatted elapsed time string, e.g. "1m 23s" or "2m".
     */
    public function elapsedDisplay(): string
    {
        if ($this->elapsedSeconds < 60) {
            return $this->elapsedSeconds . 's';
        }

        $minutes = (int) floor($this->elapsedSeconds / 60);
        $seconds = $this->elapsedSeconds % 60;

        if ($seconds === 0) {
            return $minutes . 'm';
        }

        return $minutes . 'm ' . $seconds . 's';
    }

    /**
     * Formatted token + cost summary, e.g. "1,234 tok | $0.0042".
     */
    public function usageDisplay(): string
    {
        $tok = number_format($this->tokensUsed);
        $cost = number_format($this->costUsd, 4);

        return "{$tok} tok | \${$cost}";
    }
}

File: python_port/src/sugarcrush/tui/agent_display_state.py
"""
Agent display state for the renderer's agent view.
"""

from dataclasses import dataclass


@dataclass
class AgentDisplayState:
    """
    Display state for one agent, consumed by AgentStatusBar and AgentViewPane.
    """
    name: str
    status: str  # 'working' | 'stopped'
    operation: str
    elapsed_seconds: int = 0
    tokens_used: int = 0
    cost_usd: float = 0.0

    @classmethod
    def new(
        cls,
        name: str,
        status: str,
        operation: str,
        elapsed_seconds: int = 0,
        tokens_used: int = 0,
        cost_usd: float = 0.0,
    ) -> 'AgentDisplayState':
        return cls(name, status, operation, elapsed_seconds, tokens_used, cost_usd)

    def elapsed_display(self) -> str:
        """Format elapsed time as '0s', '1m 30s', etc."""
        total = self.elapsed_seconds
        if total < 60:
            return f'{total}s'
        minutes = total // 60
        seconds = total % 60
        if minutes < 60:
            return f'{minutes}m {seconds}s'
        hours = minutes // 60
        minutes = minutes % 60
        return f'{hours}h {minutes}m'

    def usage_display(self) -> str:
        """Format token/cost usage as '0 tok | $0.0000'."""
        tokens = self.tokens_used
        cost = self.cost_usd
        return f'{tokens} tok | ${cost:.4f}'
